application/src/main/java/uk/gov/hmcts/ccd/definition/store/flyway/FlywayNoOpStrategy.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.ccd.definition.store.flyway;
+
+import org.flywaydb.core.Flyway;
+import org.springframework.boot.autoconfigure.flyway.FlywayMigrationStrategy;
+
+import java.util.stream.Stream;
+
+public class FlywayNoOpStrategy implements FlywayMigrationStrategy {
+
+    @Override
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    public void migrate(Flyway flyway) {
+        Stream.of(flyway.info().all())
+            .filter(info -> !info.getState().isApplied())
+            .findFirst()
```

Comment:
Will there be only 0 or 1 migration not applied? If so ok, but if not, better if we collect them all and throw the exception with a list of scripts, rather than just one.

Reply:
as discussed the remediation of first detected will fix all others as well. This is only happens if pipeline misses gradle task.